### PR TITLE
[ML-27512] Pass exogenous var to auto_arima

### PR DESCRIPTION
**Pull request by maddiedawson**
Modify Arima APIs to use regressors/exogenous variables during training and inference. If a model is trained with regressors, the regressors must be passed in during inference.

### CODE REVIEW

File: runtime/databricks/automl_runtime/forecast/pmdarima/model.py [hunk 1]
```diff
@@ -177,18 +183,30 @@ def _predict_impl(self, input_df: pd.DataFrame) -> pd.DataFrame:
         # Out-of-sample prediction if needed
         horizon = calculate_period_differences(self._end_ds, max(df["ds"]), self._frequency)
         if horizon > 0:
-            future_pd = self._forecast(horizon)
+            # Forward fill missing time steps
+            X_future = df[df["ds"] > self._end_ds].set_index("ds").resample(rule=OFFSET_ALIAS_MAP[self._frequency]).pad()
```
Comment (lu-wang-dl) [inline runtime/databricks/automl_runtime/forecast/pmdarima/model.py:187]: Maybe later, we could handle this in the data preprocessing step.

File: runtime/databricks/automl_runtime/forecast/pmdarima/model.py [hunk 2]
```diff
@@ -283,7 +310,11 @@ def make_future_dataframe(
         )
         return future_df
 
-    def predict_timeseries(self, horizon: int = None, include_history: bool = True) -> pd.DataFrame:
+    def predict_timeseries(
```
Comment (lu-wang-dl) [inline runtime/databricks/automl_runtime/forecast/pmdarima/model.py:313]: Should we create a new function instead of changing this one?

**Review by lu-wang-dl — COMMENTED** [review]
Overall LGTM. Left few comments and wait for fixing the tests.

File: runtime/databricks/automl_runtime/forecast/pmdarima/model.py (hunk 2 above)
Comment (maddiedawson) [reply runtime/databricks/automl_runtime/forecast/pmdarima/model.py:312]: I made the df optional

**Review by lu-wang-dl — APPROVED** [review]

**Pull request merged**